app/models/miq_request_workflow.rb

```diff
@@ -1069,7 +1069,7 @@ def allowed_storages(_options = {})
     MiqPreloader.preload(hosts, :storages => {}, :host_storages => :storage)
 
     storages = hosts.each_with_object({}) do |host, hash|
-      host.writable_storages.each { |s| hash[s.id] = s }
+      host.writable_accessible_storages.each { |s| hash[s.id] = s }
     end.values
```

Comment:
This change will help with the Provisioning UI and Automate callers to `eligible_storages`.

Unfortunately the `vmware_best_fit_least_utilized` placement method in Automate provisioning uses `writable_storages` directly.  (See [VM/Provisioning/Placement.class/__methods__/vmware_best_fit_least_utilized.rb](https://github.com/ManageIQ/manageiq-content/blob/master/content/automate/ManageIQ/Infrastructure/VM/Provisioning/Placement.class/__methods__/vmware_best_fit_least_utilized.rb#L45))

@tinaafitz As part of fixing this BZ we should look into changing these calls to use `allowed_storages` instead.